galasa-inttests-parent/dev.galasa.inttests/src/main/java/dev/galasa/inttests/compilation/AbstractCompilationLocal.java

```diff
@@ -0,0 +1,411 @@
+import dev.galasa.core.manager.Logger;
+import dev.galasa.core.manager.RunName;
+import dev.galasa.core.manager.TestProperty;
+import dev.galasa.framework.spi.ResourceUnavailableException;
+import dev.galasa.galasaecosystem.IGenericEcosystem;
+import dev.galasa.http.HttpClient;
+import dev.galasa.http.IHttpClient;
+import dev.galasa.ipnetwork.IpNetworkManagerException;
+import dev.galasa.java.ubuntu.IJavaUbuntuInstallation;
+import dev.galasa.linux.ILinuxImage;
+import dev.galasa.linux.LinuxManagerException;
+
+public abstract class AbstractCompilationLocal {
+    
+    @RunName
+    public String         runName;
+    
+    @Logger
+    public Log              logger;
+    
+    @HttpClient
+    public IHttpClient      client;
+    
+    private String          prefix = "dev.galasa.simbank";
+    private String          testProjectName = prefix + ".tests";
+    private String          managerProjectName = prefix + ".manager";
+    
+    private Path            simplatformParentDir;
+    private Path            gradleBin;
+    
+    @TestProperty(prefix = "gradle.zip",suffix = "location", required = true)
+    public String           gradleZipLocation;
+    
+    public String           gradleZipVersion;
+    
+    String                  javaHomeCommand;
+    
+    /*
+     * Initialise all resources needed for testing.
+     * 
+     */
+    @BeforeClass
+    public void setupTest() throws ResourceUnavailableException, IOException, LinuxManagerException, IpNetworkManagerException {
+        javaHomeCommand = "export JAVA_HOME=" + getJavaInstallation().getJavaHome();
+        
+        gradleZipVersion = getGradleVersion();
+        simplatformParentDir = setupSimPlatform();
+        gradleBin = installGradle();
+    }
+    
+    /*
+     * Extracts the version from the gradle zip path
+     */
+    
+    public String getGradleVersion() {
+        final String regex = "gradle-(\\d+\\.\\d+\\.\\d+)-bin\\.zip";
+
+        final Pattern pattern = Pattern.compile(regex, Pattern.MULTILINE);
+        final Matcher matcher = pattern.matcher(gradleZipLocation);
+        matcher.find();
+        gradleZipVersion = matcher.group(1);
+        
+        return gradleZipVersion;
+    }
+
+    /*
+     * Download simplatform (github main branch), unzipp simplatform, then structure 
+     * and edit the directory and files it in such a way that they are usable by gradle.
+     * 
+     * @return  remoteDirectory     the parent directory of the structured, edited, and usable simplatform project.
+     * 
+     */
+    private Path setupSimPlatform() throws ResourceUnavailableException, IOException, LinuxManagerException {
+        logger.info("Downloading simplatform repository archive - main branch.");
+        Path simplatformZip = downloadHttp("https://github.com/galasa-dev/simplatform/archive/main.zip");
+        logger.info("Unzipping simplatform repository archive");
+        Path simplatformDir = unpackArchiveToTemp(simplatformZip);
```

Comment:
You should be copying the zip to the test machine,  before you unzip.  All the work should be happening there,  not in the docker container that is the test engine.   will reduce network traffic

Reply:
Files are now unpackaged on the remote image.